internal/provider/models/processors/set_timestamp.go

```diff
@@ -0,0 +1,156 @@
+package processors
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	. "github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/client"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/provider/models/modelutils"
+)
+
+const SET_TIMESTAMP_PROCESSOR_TYPE_NAME = "set_timestamp"
+const SET_TIMESTAMP_PROCESSOR_NODE_NAME = "set-timestamp"
+
+type SetTimestampProcessorModel struct {
+	Id           StringValue `tfsdk:"id"`
+	PipelineId   StringValue `tfsdk:"pipeline_id"`
+	Title        StringValue `tfsdk:"title"`
+	Description  StringValue `tfsdk:"description"`
+	Inputs       ListValue   `tfsdk:"inputs"`
+	GenerationId Int64Value  `tfsdk:"generation_id"`
+	Parsers      ListValue   `tfsdk:"parsers" user_config:"true"`
+}
+
+var parserAttrTypes = map[string]attr.Type{
+	"field":                   StringType{},
+	"timestamp_format":        StringType{},
+	"custom_timestamp_format": StringType{},
+}
+
+var SetTimestampProcessorResourceSchema = schema.Schema{
+	Description: "Parse a list of fields using a chosen time format. First match will override the default timestamp of the event.",
+	Attributes: ExtendBaseAttributes(map[string]schema.Attribute{
+		"parsers": schema.ListNestedAttribute{
+			Required:    true,
+			Description: "The list of fields and parsers to use in order of priority, short-circuiting on the first successful match.",
+			Validators: []validator.List{
+				listvalidator.SizeAtLeast(1),
+			},
+			NestedObject: schema.NestedAttributeObject{
+				Attributes: map[string]schema.Attribute{
+					"field": schema.StringAttribute{
+						Required:    true,
+						Description: "The field whose value will be parsed.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"timestamp_format": schema.StringAttribute{
+						Required:    true,
+						Description: "The timestamp format.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"custom_timestamp_format": schema.StringAttribute{
+						Optional:    true,
+						Description: "Custom timestamp format, according to strftime.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
```

Comment:
If you have to give the full format for either of these, is it worth having 2 fields? Should it just be "timestamp_format" and if it's not a "pre-defined" one, have that imply that the type is custom?  There's no apparent way for an end-user to know what the pre-defined formats are, that aren't "custom"

Reply:
I just matched what we did in pipeline-service.   I could do that, but I would have to copy (and subsequently) keep in sync all of the timestamps we have in pipeline-service.

Reply:
I mean we do that in a bunch of other places, to me, this is at the least, confusing UX.  On a stressful day I would yell at my screen about it :D

Reply:
Or perhaps the TF version of the processor could always use "custom", since the user has to come up with the format string themselves anyways

Reply:
I like that idea.  That might make more sense to the TF user as well (and I will add it to the documentaiton.